### PR TITLE
Enable caching of compiled guvectorize functions

### LINKED ISSUE #363: Importing sgkit is slow
It takes ~5 seconds on my machine - most of this is numba compilation. I notice this a lot when running tests.

### PR DESCRIPTION
**Pull request by tomwhite**
Fixes #363

```
$ time python -c "import sgkit"
real	0m5.481s
user	0m5.960s
sys	0m0.527s
$ time python -c "import sgkit"
real	0m1.982s
user	0m2.520s
sys	0m0.471s
```

BTW @aktech I noticed that the `guvectorize` metric functions don't have `nopython=True` - do you think we should add it? (I could do it in this PR.)

### CODE REVIEW

File: sgkit/stats/popgen.py [hunk 1]
```diff
@@ -121,9 +121,7 @@ def diversity(
 
 # c = cohorts, k = alleles
 @guvectorize(  # type: ignore
```
Comment (jeromekelleher) [inline sgkit/stats/popgen.py:123]: Is there any good reason not to have ``cache=True`` for our use-case? Perhaps it's worth us having our own local ``guvectorize`` decorator that does this by default?

This is the sort of thing that's hard to test for and easy to forget, so maybe it's worth a little indirection.

**Review by jeromekelleher — APPROVED** [review]
LGTM, minor query

**Comment by aktech** [comment]
> BTW @aktech I noticed that the guvectorize metric functions don't have nopython=True - do you think we should add it? (I could do it in this PR.)

That will be a good addition, I think.

**Comment by aktech** [comment]
This is a really nice improvement, by the way. 👍

**Review by aktech — APPROVED** [review]

File: sgkit/stats/popgen.py (hunk 1 above)
Comment (tomwhite) [reply sgkit/stats/popgen.py:123]: Maybe? It's not hard to add and now we have the convention started it is likely to be propagated in our code.

Comment (tomwhite) [reply sgkit/stats/popgen.py:123]: Opened #371

**Pull request merged**